### PR TITLE
Feature/ohlcv in dataframe

### LINKED ISSUE #18: OHLCV data / indicators to DataFrame
**(Together with Tijs)**

Currently, all our OHLCV data and indicators are stored in either an object array or dictionary. This is not really useful for TA-Lib / calculating purposes (since TA-Lib requires numpy array as input). We need to put all data inside a dataframe, and make this work application-wide. This is necessary for the strategies to work.

### PR DESCRIPTION
**Pull request by noudcorten**
# Fixes

Fixes #18 #35 #41 

# What has been changed?

* OHLCV data is now stored in pandas dataframes
* strategy.py can now correctly be accessed
* dataframes are stored in .json files with nice formatting
* backtesting period now covers correct time period (HOTFIX)
* existing candle data is re-used
* removed unnecessary files (previously used for OHLCV data)
* code optimalisation / improved readability

# Examples

# Actions to be performed before this can be merged

- [ ] PR depends on #
- [ ] Script to be run

# Security Measures

* Added check if dataframes for all pairs have equal lengths

# Potential Issues/Future considerations

* Method for backtesting is significantly slower (takes around 13-14 seconds for standard config.json). Has nothing to do with bad implementation, just requesting data from a dataframe is slow.

# Additional Info

* Tested another implementation in branch: feature/ohlcv-in-dataframe-v2. This implementation only made a dataframe just before it had to be used in strategy.py. This implementation works, but is even MORE slow than current implementation.
* Contains HOTFIX

### CODE REVIEW

File: backtesting/backtesting.py [hunk 1]
```diff
@@ -48,16 +48,17 @@ def start_backtesting(self, data, backtesting_from, backtesting_to) -> None:
...
-            for tick in data[pair]:
-                data_per_tick[tick.time].append(tick)
+        pairs = list(data.keys())
+        ticks = list(data[pairs[0]].index.values)
+        
+        for i, tick in enumerate(ticks):
+            for pair in pairs:
+                # Get df for current pair and retrieve ohlcv for current tick
+                pair_df = data[pair]
+                ohlcv_tick = pair_df.loc[tick].copy()
```
Comment (dema-trading-ai) [inline backtesting/backtesting.py:58]: Cool, very clean coding. I like your codingstyle.

File: backtesting/backtesting.py [hunk 2]
```diff
@@ -48,16 +48,17 @@ def start_backtesting(self, data, backtesting_from, backtesting_to) -> None:
...
 
-        ticks_passed_per_pair = defaultdict(self.default_empty_array_dict)
-        for tick in data_per_tick:
-            for pair_tick in data_per_tick[tick]:
-                self.trading_module.tick(pair_tick, ticks_passed_per_pair[pair_tick.pair])
-                ticks_passed_per_pair[pair_tick.pair].append(pair_tick)
+                # Get passed ticks and pass to trading module
+                self.trading_module.tick(ohlcv_tick, pair_df[:i])
```
Comment (dema-trading-ai) [inline backtesting/backtesting.py:61]: Clean.

**Review by dema-trading-ai — APPROVED** [review]
I like your coding style! Very neat & tidy & clean!

File: backtesting/backtesting.py (hunk 1 above)
Comment (noudcorten) [reply backtesting/backtesting.py:58]: Thanks! That's really nice to hear :)

**Pull request merged**